core/dbt/parser/sources.py

```diff
@@ -270,10 +268,11 @@ def _generate_source_config(self, fqn: List[str], rendered: bool, project_name:
 
         return generator.calculate_node_config(
             config_call_dict={},
-            fqn=fqn,
+            fqn=target.fqn,
             resource_type=NodeType.Source,
-            project_name=project_name,
+            project_name=target.package_name,
             base=False,
+            patch_config_dict=target.source.config,
```

Comment:
We need access to both:
- `target.source.config` = config defined at the source level
- `target.table.config` = config defined at the source table level

I think there are two options here:
- Merge those configs together (preferring table-level over source-level), before passing into `patch_config_dict`
- Modify `calculate_node_config` to accept multiple "levels" of patches, so that all calculation happens in that same method: source table > source > project-level config